encoder/csv.go

```diff
@@ -54,31 +63,16 @@ func NewCSVItemWriter(sourceConfig map[string]any, data io.WriteCloser, batchInf
 			if batchInfo != nil {
 				if batchInfo.IsStartBatch {
 					// create comma separated header row from writer.columns
-					headerRow := strings.Join(writer.columns, writer.separator)
-					_, err := data.Write([]byte(headerRow)) // write the header for the csv-file
-					// set headerWritten to true, so we don't write the header again
-					writer.headerWritten = true
+					//headerRow := strings.Join(writer.columns, writer.separator)
+					err := writer.encoder.Write(writer.columns) // write the header for the csv-file
 					if err != nil {
 						return nil, err
 					}
-				} else {
-					writer.firstItemWritten = true
 				}
 			}
```

Comment:
When we discussed this before we talked about some code here, I thought, for when its incremental?